scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,490 @@
+version: "1.8"
+class: $tMetricGroup
+name: AddMetricExp
+decl: static
+details:
+    - "Reasons for failing to add the metric could be queried using pErrorString"
+params:
+    - type: $t_metric_group_handle_t 
+      name: hMetricGroup
+      desc: "[in] Handle of the metric group"
+    - type: $t_metric_handle_t
+      name: hMetric
+      desc: "[in] Metric to be added to the group."
+    - type: "uint32_t*"
+      name: pErrorStringSize
+      desc: "[in,out][optional] Size of the error string to query, if an error was reported during adding the metric handle."
+    - type: "char*"
+      name: pErrorString
+      desc: |
+            [in,out][optional][range(0, *pErrorStringSize)] Error string.
+            if *pErrorStringSize is less than the length of the error string available, then driver shall only retrieve that length of error string.
+returns:
+    - $X_RESULT_ERROR_INVALID_ARGUMENT:
+        - "If a Metric handle from a pre-defined metric group is requested to be added"
+    - $X_RESULT_ERROR_UNSUPPORTED_FEATURE:
+        - "If the Metric group handle is already activated"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Remove a metric from the metric group handle created using $tMetricGroupCreateExp."
+version: "1.8"
+class: $tMetricGroup
+name: RemoveMetricExp
+decl: static
+details:
+    - "Remove an already added metric handle from the metric group"
+params:
+    - type: $t_metric_group_handle_t
+      name: hMetricGroup
+      desc: "[in] Handle of the metric group"
+    - type: $t_metric_handle_t
+      name: hMetric
+      desc: "[in] Metric handle to be removed from the metric group."
+returns:
+    - $X_RESULT_ERROR_INVALID_ARGUMENT:
+        - "If trying to remove a metric not previously added to the metric group"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Finalizes a created metric group using $tMetricGroupCreateExp, so that it can be activated."
+version: "1.8"
+class: $tMetricGroup
+name: FinalizeExp
+decl: static
+details:
+    - "Finalize the metric group."
+    - "This is a necessary step for the created metric group, before activation."
+    - "Add / Remove of metrics is possible after Finalize. However, a call to $tMetricGroupFinalizeExp is necessary after modifying the metric group."
+params:
+    - type: $t_metric_group_handle_t
+      name: hMetricGroup
+      desc: "[in] Handle of the metric group"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Destroy a metric group created using $tMetricGroupCreateExp."
+version: "1.8"
+class: $tMetricGroup
+name: DestroyExp
+decl: static
+params:
+    - type: $t_metric_group_handle_t 
+      name: hMetricGroup
+      desc: "[in] Handle of the metric group to destroy"
+returns:
+    - $X_RESULT_ERROR_INVALID_ARGUMENT:
+        - "If trying to destroy a pre-defined metric group"
```

Comment:
- is it allowed to destroy a metric group that contains metrics? if yes, need to explicitly clarify it destroys the metrics 
- what if the metric group is active? 
-